### PR TITLE
Fixed - https://wso2.org/jira/browse/APPM-907

### PR DESCRIPTION
**Pull request by sajithab**

### CODE REVIEW

File: features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/apis/user.jag [hunk 1]
```diff
@@ -139,6 +139,24 @@ require('/modules/store.js').exec(function (ctx) {
                     message: message
                 });
                 return;
+
+            case 'emailLogin':
+                try {
+                    var isEnableEmailUsername = storeHostObj.isEnableEmailUsername(null);
```
Comment (dinusha-dilrukshi) [inline features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/apis/user.jag:145]: Why do we pass null here ? Shouldn't it be the username ?

File: features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/apis/user.jag [hunk 2]
```diff
@@ -139,6 +139,24 @@ require('/modules/store.js').exec(function (ctx) {
...
+                    if (isEnableEmailUsername) {
+                        isEmailUsername = true;
+                    }
+                } catch (e) {
+                    log.error(e);
```
Comment (jenananthan) [inline features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/apis/user.jag:150]: log error with custom error message e.g log.error("custom error message" ,e)

File: features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/themes/store/js/navigation.js [hunk 3]
```diff
@@ -21,26 +21,26 @@ $(function() {
...
-            		dataType: 'json'
+                type: 'POST',
+                url: '/apis/user/login',
+                data: JSON.stringify({
+                    username: username,
+                    password: password
+                }),
+                success: function (data) {
+                    if (!data.error) {
+                var assetId = $('#modal-login').data('value');
```
Comment (jenananthan) [inline features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/themes/store/js/navigation.js:32]: formatting issue

File: features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/apis/user.jag [hunk 4]
```diff
@@ -8,16 +8,16 @@ var AuthService=require('../extensions/assets/webapp/services/authentication.js'
...
+    log = new Log(),
+    user = require('store').user,
+    matcher = new URIMatcher(request.getRequestURI()),
+    userDetails={},
+    parameters,
+    isEmailUsername = false;
 
-   var authenticator=new AuthService.Authenticator();
-   authenticator.init(jagg,session);
+    var authenticator=new AuthService.Authenticator();
```
Comment (tgtshanika) [inline features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/apis/user.jag:19]: Format code in line no 15,19 ..

Comment (sajithab) [inline features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/apis/user.jag:19]: Fixed https://github.com/wso2/carbon-appmgt/pull/141

File: features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/apis/user.jag (hunk 1 above)
Comment (sajithab) [inline features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/apis/user.jag:145]: We do not need username in here. This is for getting the carbon.xml email username config value.
fixed https://github.com/wso2/carbon-appmgt/pull/141

File: features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/apis/user.jag (hunk 2 above)
Comment (sajithab) [inline features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/apis/user.jag:150]: Custom error will display by using the CarbonUtils class if error happened.

File: features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/themes/store/js/navigation.js (hunk 3 above)
Comment (sajithab) [inline features/appmgt/org.wso2.carbon.appmgt.store.feature/src/main/resources/store/themes/store/js/navigation.js:32]: Fixed https://github.com/wso2/carbon-appmgt/pull/141

**Pull request merged**